config/src/config/execution_config.rs

```diff
@@ -56,13 +56,41 @@ impl Default for ExecutionConfig {
 impl ExecutionConfig {
     pub fn load(&mut self, root_dir: &RootPath) -> Result<(), Error> {
         if !self.genesis_file_location.as_os_str().is_empty() {
-            let path = root_dir.full_path(&self.genesis_file_location);
-            let mut file = File::open(&path).map_err(|e| Error::IO("genesis".into(), e))?;
+            // Ensure the genesis file exists
+            let genesis_path = root_dir.full_path(&self.genesis_file_location);
+            if !genesis_path.exists() {
+                return Err(Error::Unexpected(format!(
+                    "The genesis file could not be found! Ensure the given path is correct: {:?}",
+                    genesis_path.display()
+                )));
+            }
+
+            // Open the genesis file and read the bytes
+            let mut file = File::open(&genesis_path).map_err(|error| {
+                Error::Unexpected(format!(
+                    "Failed to open the genesis file: {:?}. Error: {:?}",
+                    genesis_path.display(),
+                    error
+                ))
+            })?;
             let mut buffer = vec![];
-            file.read_to_end(&mut buffer)
-                .map_err(|e| Error::IO("genesis".into(), e))?;
-            let data = bcs::from_bytes(&buffer).map_err(|e| Error::BCS("genesis", e))?;
-            self.genesis = Some(data);
+            file.read_to_end(&mut buffer).map_err(|error| {
+                Error::Unexpected(format!(
+                    "Failed to read the genesis file into a buffer: {:?}. Error: {:?}",
+                    genesis_path.display(),
+                    error
+                ))
+            })?;
+
+            // Deserialize the genesis file and store it
+            let genesis = bcs::from_bytes(&buffer).map_err(|error| {
+                Error::Unexpected(format!(
+                    "Failed to BCS deserialize the genesis file: {:?}. Error: {:?}",
```

Comment:
ditto

Reply:
The file has already been read at this point. It's a deserialization error.